src/Toolkit/Toolkit.Maui/SymbolDisplay/SymbolDisplay.cs

```diff
@@ -120,6 +148,12 @@ private async Task UpdateSwatchAsync()
             try
             {
                 var scale = GetScaleFactor();
+                _lastScaleFactor = scale;
+
+                if (double.IsNaN(scale))
+                {
+                    throw new Exception();
+                }
```

Comment:
Since this is not exceptional, just return here instead of throwing (and generally we don't throw the base exception type and with no message)